libnmstate/nm/device.py

```diff
@@ -173,62 +173,39 @@ def delete_device(nmdev):
 
 def _safe_delete_device_async(nmdev):
     mainloop = nmclient.mainloop()
-    user_data = mainloop, nmdev
+    user_data = mainloop, nmdev, nmdev.get_iface()
```

Comment:
As time of this function been called, `nmdev.get_iface()` could be None.

You may try `assert nmdev.get_iface()` in the repeat calls of bond creation and deletion.

Reply:
The interface is only used for logging, so this does not add extra errors AFAIU. Also AFAIU this code is called before the mainloop is started, therefore it is strange that the interface could be `None`. IMHO the proper solution would be to pass  a static context that contains the information about which state is processed here (the original state, the desired state, the interface that is changed, ...) instead of only using changing NM objects. But this would be a bigger change in Nmstate.

Reply:
The interface became None as previous mainloop run might enter a partial updated `NM.Device`. Normally, profile deletion will delete the soft interface as well, so then the mainloop of profile deletion quit, the `NM.Device` might be in the middle of deletion.